apps/backend/src/services/audit_service.py

```
@@ -0,0 +1,443 @@
+                top_data_types=top_data_types,
+                top_errors=top_errors,
+                period_start=period_start,
+                period_end=period_end,
+            )
+
+        except Exception as e:
+            logger.error(f"Error getting audit metrics: {e}")
+            raise DatabaseOperationError(f"Failed to get audit metrics: {str(e)}", "get_audit_metrics")
+
+    async def get_performance_report(
+        self, device_id: Optional[UUID] = None, period: str = "last_24h"
```

Ruff (0.12.2)

Use `X | None` for type annotations

Convert to `X | None`

(UP045)